apps/researcher/src/lib/community.test.ts

```diff
@@ -92,3 +97,115 @@ describe('isAdmin', () => {
     expect(isAdmin(memberships)).toEqual(false);
   });
 });
+
+describe('sortCommunities', () => {
+  const communities = [
+    {
+      id: 'community2',
+      name: 'Community 2',
+      slug: 'community-2',
+      imageUrl: 'https://example.com/image.png',
+      createdAt: 1690000000000,
+    },
+    {
+      id: 'community1',
+      name: 'Community 1',
+      slug: 'community-1',
+      imageUrl: 'https://example.com/image.png',
+      createdAt: 1600000000000,
+    },
+    {
+      id: 'community4',
+      name: 'Community 4',
+      slug: 'community-4',
+      imageUrl: 'https://example.com/image.png',
+      createdAt: 1680000000000,
+    },
+    {
+      id: 'community3',
+      name: 'Community 3',
+      slug: 'community-3',
+      imageUrl: 'https://example.com/image.png',
+      createdAt: 1650000000000,
+    },
+    {
+      id: 'community5',
+      name: 'Community 5',
+      slug: 'community-5',
+      imageUrl: 'https://example.com/image.png',
+      createdAt: 1670000000000,
+    },
+  ];
+  it('sorts communities by name ascending', () => {
+    const sortedCommunities = sortCommunities(
+      communities,
+      CommunitySortBy.NameAsc
+    );
+    const sortedCommunitiesIds = sortedCommunities.map(
+      community => community.id
+    );
+    expect(sortedCommunitiesIds).toEqual([
+      'community1',
+      'community2',
+      'community3',
+      'community4',
+      'community5',
+    ]);
+  });
+
+  it('sorts communities by name descending', () => {
+    const sortedCommunities = sortCommunities(
+      communities,
+      CommunitySortBy.NameDesc
+    );
+    const sortedCommunitiesIds = sortedCommunities.map(
+      community => community.id
+    );
+    expect(sortedCommunitiesIds).toEqual([
+      'community5',
+      'community4',
+      'community3',
+      'community2',
+      'community1',
+    ]);
+  });
+
+  it('sorts communities by creation date descending', () => {
+    const sortedCommunities = sortCommunities(
+      communities,
+      CommunitySortBy.CreatedAtDesc
+    );
+    const sortedCommunitiesIds = sortedCommunities.map(
+      community => community.id
+    );
+    expect(sortedCommunitiesIds).toEqual([
+      'community2',
+      'community4',
+      'community5',
+      'community3',
+      'community1',
+    ]);
+  });
+
+  it('works with an empty array', () => {
```

Comment:
1. Instead of: `array`: `list` (to use the same words as on e.g. line 195)?
2. Thinking out load: what is the purpose of this test? 'works with' is a bit vague - what does it mean? For example: do you want to test whether there are no side-effects when passing an empty list (e.g. to test that the function doesn't add data)?

Reply:
I think I'm testing that it won't error. However, I did not have a well-thought-out reason for this test. So I will remove it.